Sources/ComposedUI/CollectionView/CollectionCoordinator.swift

```diff
@@ -73,25 +73,25 @@ open class CollectionCoordinator: NSObject {
         prepareSections()
 
         delegateObserver = collectionView.observe(\.delegate, options: [.initial, .new]) { [weak self] collectionView, _ in
-            guard collectionView.delegate !== self else { return }
+            guard !(collectionView.delegate is CollectionCoordinator) else { return }
```

Comment:
This does't satisfy the actual requirement.
It can't be just ANY coordinator, it needs to be this one, since this one owns the mappings, etc...

Reply:
I get the reason for the original code, but I'm not sure why the new code would be a problem (although I'm sure it could be, I just don't know the scenario to test that would show the problem).

I think since the crash this is working around is fixed via `replace(sectionProvider:)` the improvement to be made is not strictly necessary for normal function, but rather catching this issue earlier and aiding with diagnosing it.

Might as well close for now and maybe look at tweaking in the future.

Reply:
I get your point. Definitely need to better understand your issue, cause atm I don't.

I can easily explain the issue with this one though. The line just below `collectionView.delegate = self` would cause KVO to retrigger, thus ending up in an infinite loop. So this check is just to see whether we even need to set the delegate, if not, we can avoid it, thus avoiding the loop.

Hence why your solution would 'work' except it wouldn't guarantee the delegate is `this` object, rather `any` coordinator. Which might not be correct and could cause other issues downstream related to the mappings for example.

Would you mind opening a new Issue to track your problem and describe it as best as possible, so I can review and maybe provide additional context to discover the right solution?

Thanks Joe.